src/qcodes/instrument_drivers/Keysight/private/Keysight_344xxA_submodules.py

```diff
@@ -1212,14 +1212,32 @@ def read(self) -> np.ndarray:
         raw_vals: str = self.ask("READ?")
         return _raw_vals_to_array(raw_vals)
 
+    def _ask_with_sense_function(self, cmd: str) -> str:
+        function = self.sense_function.get_latest.get_raw().strip("\"")
```

Comment:
this is a more correct way of getting raw value from cache
```suggestion
        function = self.sense_function.cache.raw_value.strip("\"")
```

Reply:
The unit tests are failing here since `cache.raw_value` is None. (I don't know if this was an issue before because the CI required approval to run.) Is my understanding correct that the cache will only be populated after `self.sense_function()` is executed at least once?

Reply:
I agree this is a bug in the cache implementation that there is no way to do this. I am also pusled why the code did not error before. For now, I have pushed code to call get on the cache before using the raw value. That should ensure its always up to date